sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-evaluation-result-processor/cas-active-supplier-found-processor.ts

```diff
@@ -0,0 +1,103 @@
+import { Repository } from "typeorm";
+import { CASAuthDetails } from "@sims/integrations/cas";
+import { CASEvaluationResultProcessor, ProcessorResult } from ".";
+
+/**
+ * Process the active supplier information found on CAS.
+ */
+@Injectable()
+export class CASActiveSupplierFoundProcessor extends CASEvaluationResultProcessor {
+  constructor(
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    super();
+  }
+
+  /**
+   * Update student supplier based on the supplier information found on CAS.
+   * @param casSupplier student supplier information from SIMS.
+   * @param evaluationResult evaluation result to be processed.
+   * @param _auth authentication token needed for possible
+   * CAS API interactions.
+   * @param summary current process log.
+   * @returns processor result.
+   */
+  async process(
+    casSupplier: CASSupplier,
+    evaluationResult: CASEvaluationResult,
+    _auth: CASAuthDetails,
+    summary: ProcessSummary,
+  ): Promise<ProcessorResult> {
+    if (evaluationResult.status !== CASEvaluationStatus.ActiveSupplierFound) {
+      throw new Error("Incorrect CAS evaluation result processor selected.");
+    }
+    summary.info("Active CAS supplier found.");
+    try {
+      const supplierToUpdate = evaluationResult.activeSupplier;
+      let supplierAddressToUpdate: SupplierAddress = null;
+      if (evaluationResult.matchedAddress) {
+        // If no matched address sites was found, the address information will be
+        // persisted as null and the supplier isValid will be set to false for now
+        // till the CAS API call for "create site" is implemented.
+        // TODO: implement site creation on CAS.
+        const address = evaluationResult.matchedAddress;
+        supplierAddressToUpdate = {
+          supplierSiteCode: address.suppliersitecode,
+          addressLine1: address.addressline1,
+          addressLine2: address.addressline2,
+          city: address.city,
+          provinceState: address.province,
+          country: address.country,
+          postalCode: address.postalcode,
+          status: address.status,
+          siteProtected: address.siteprotected,
+          lastUpdated: new Date(address.lastupdated),
+        };
+      }
+      const now = new Date();
+      const systemUser = this.systemUsersService.systemUser;
+      const updateResult = await this.casSupplierRepo.update(
+        {
+          id: casSupplier.id,
+        },
+        {
+          supplierNumber: supplierToUpdate.suppliernumber,
+          supplierName: supplierToUpdate.suppliername,
+          status: supplierToUpdate.status,
+          supplierProtected: supplierToUpdate.supplierprotected === "Y",
+          lastUpdated: new Date(supplierToUpdate.lastupdated),
+          supplierAddress: supplierAddressToUpdate,
+          supplierStatus: SupplierStatus.Verified,
```

Comment:
create site api call that needs to be implemented later will be created using a processor under the same umbrella of processors, correct? If that is the case, then along with `isValid` to false, don't we also need to set this to `PendingAddressVerification` or some `Pending` status so that it can be picked up again in future as a CAS supplier record for which the site information needs to be created.

Reply:
`PendingAddressVerification` is intended to be used when the SIMS student address changes.
For now, there are no requirements to set it as pending since there is no process to check it.

Reply:
ok